### PR TITLE
Add GSOC 2018 page

### PR DESCRIPTION
**Pull request by ericholscher**
This is our initial attempt to build a set of projects for GSOC.
It isn't an exhaustive list,
and could use more ideas!

### CODE REVIEW

File: docs/gsoc.rst [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+Read the Docs is hoping to participate in the Google Summer of Code in 2018.
+This page will contain all the information for students and anyone else interested in helping.
+
+Required Skills
+---------------
+
+Incoming students will need the following skills:
+
+* Intermediate python programming
```
Comment (safwanrahman) [inline docs/gsoc.rst]: I think `django` and `docker` should be added

File: docs/gsoc.rst [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+* Familiarity with Markdown, reStructuredText, or some other plain text markup language
+* Familiarity with git, or some other source control
+* Ability to set up your own development environment for Read the Docs
+* An interest in documentation and improving open source documentation tools would be great too!
```
Comment (safwanrahman) [inline docs/gsoc.rst]: I think there should be something like
`Having some JS/CSS knowledge would be awesome things to have`

File: docs/gsoc.rst (hunk 1 above)
Comment (ericholscher) [reply docs/gsoc.rst]: Agreed re: Django -- there are projects that can be done without docker, so I'm less worried about that.

**Comment by safwanrahman** [comment]
I think its ready to go! r+

**Comment by ericholscher** [comment]
@Blendify look good?

**Comment by ericholscher** [comment]
🎆

**Pull request merged**